ext/reflection/php_reflection.c

```diff
@@ -2630,6 +2661,46 @@ ZEND_METHOD(reflection_parameter, getClass)
 }
 /* }}} */
 
+/* {{{ proto public bool ReflectionParameter::hasType()
+   Rethern whether parameter has a type hint */
+ZEND_METHOD(reflection_parameter, hasType)
+{
+	reflection_object *intern;
+	parameter_reference *param;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+	GET_REFLECTION_OBJECT_PTR(param);
+
+	RETVAL_BOOL((param->fptr->type == ZEND_INTERNAL_FUNCTION ?
+		((zend_internal_arg_info*)param->arg_info)->type_hint :
+		param->arg_info->type_hint) != 0);
+}
+/* }}} */
+
+/* {{{ proto public string ReflectionParameter::getType()
```

Comment:
Return type here should be `ReflectionType` rather than `string`. Same for `getReturnType()`.